shared/common-adapters/modal2.tsx

```diff
@@ -0,0 +1,275 @@
+import * as React from 'react'
+import * as Styles from '../styles'
+import {LayoutChangeEvent} from 'react-native'
+import ScrollView from './scroll-view'
+import {Box2, Box} from './box'
+import BoxGrow from './box-grow'
+import Text from './text'
+import {useTimeout} from './use-timers'
+
+const Kb = {
+  Box,
+  Box2,
+  BoxGrow,
+  ScrollView,
+  Text,
+  useTimeout,
+}
+
+type HeaderProps = {
+  hideBorder?: boolean
+  icon?: React.ReactNode // above center
+  leftButton?: React.ReactNode
+  rightButton?: React.ReactNode
+  subTitle?: React.ReactNode // center; be sure to lineClamp any long / dynamic strings
+  subTitleAbove?: boolean
+  title?: React.ReactNode // center; be sure to lineClamp any long / dynamic strings
+  style?: Styles.StylesCrossPlatform
+}
+
+type FooterProps = {
+  content: React.ReactNode
+  hideBorder?: boolean
+  style?: Styles.StylesCrossPlatform
+}
+
+type Props = {
```

Comment:
allowOverflow, onClose, mobileStyle, noScrollView, scrollViewRef are unused. what's plumbing a callback to `onClose` like with this?

Reply:
(Seems ok to skip the scrollview that was builtin in the other one)

Reply:
I think we'd mostly not have an Onclose callback. 98% of the time it just navs up. If you want to know more you can use the nav events hook or watch on nav changed event

Reply:
Some places override with a `clearModals`, and in the add to team wizard we need to do that + clear a bunch of state. Both possible with using the navChanged, but also would probably be easy to plumb through navigation options since those are also just simple action dispatches (don't depend on screen state)